lib/get-selenium-status-url.js

```diff
@@ -1,36 +1,71 @@
-module.exports = getSeleniumStatusUrl;
-
 var URI = require('urijs');
+var PROCESS_TYPES = exports.PROCESS_TYPES = {
+  STANDALONE: 0,
+  GRID_HUB: 1,
+  GRID_NODE: 2
+};
 
-function getSeleniumStatusUrl(seleniumArgs) {
-  var port = 4444;
-  var hostname = 'localhost';
-  var portArg = seleniumArgs.indexOf('-port');
-  var hubArg = seleniumArgs.indexOf('-hub');
+exports.getRunningProcessType = function(seleniumArgs) {
   var roleArg = seleniumArgs.indexOf('-role');
+  var role = (roleArg !== -1) ? seleniumArgs[roleArg + 1] : undefined;
 
-  if (hubArg !== -1) {
-    hubURI = new URI(seleniumArgs[hubArg + 1]);
-    hostname = hubURI.hostname() || hostname;
-    port = hubURI.port() || port;
-  }
+  if (roleArg === -1) return PROCESS_TYPES.STANDALONE;
+  else if (role === 'hub') return PROCESS_TYPES.GRID_HUB;
+  else if (role === 'node') return PROCESS_TYPES.GRID_NODE;
+  else return undefined;
+}
+
+exports.getSeleniumStatusUrl = function(seleniumArgs) {
+  var processType = this.getRunningProcessType(seleniumArgs);
+  var portArg = seleniumArgs.indexOf('-port');
+  var port = (portArg !== -1) ? seleniumArgs[portArg + 1] : undefined;
 
-  if (roleArg !== -1) {
-    switch(seleniumArgs[roleArg + 1]){
-      case 'node':
-        if(hubArg === -1){
-          port = portArg !== -1 ? seleniumArgs[portArg + 1] : 5555;
-        }
-        return 'http://' + hostname + ':' + port + '/wd/hub/status';
-      case 'hub':
-        port = portArg !== -1 ? seleniumArgs[portArg + 1] : 4444;
-        return 'http://localhost:' + port + '/api/grid/hub/';
-    }
+  var statusURI = new URI('http://localhost');
```

Comment:
What I do not get is why it would always be "localhost" here? Can't the hub be started and only listen on another interface?